File: src/main/java/ux/BasicPanel.java
package ux;

public class BasicPanel extends Panel {
    protected BasicPanel(Builder builder) {
        super(builder);
    }

    @Override
    protected void panelSubclassDraw() {
        ;
    }

    public static class Builder extends Panel.Builder<Builder> {

        public Builder(PGraphicsSupplier graphicsSupplier, AlignHorizontal alignHorizontal, AlignVertical vAlign, int width, int height) {
            super(graphicsSupplier, alignHorizontal, vAlign, width, height);
        }

        @Override
        public BasicPanel build() {
            return new BasicPanel(this);
        }

        @Override
        protected Builder self() {
            return this;
        }
    }
}

File: src/main/kotlin/patterning/panel/BasicPanel.kt
package patterning.panel

import patterning.Canvas
import patterning.DrawingInformer

class BasicPanel private constructor(builder: Builder) : Panel(builder) {
    override fun panelSubclassDraw() {
    }
    
    class Builder(
        drawingInformer: DrawingInformer,
        canvas: Canvas,
        hAlign: AlignHorizontal,
        vAlign: AlignVertical,
        width: Int,
        height: Int
    ) : Panel.Builder(
        drawingInformer, canvas,hAlign, vAlign, width, height
    ) {
        override fun build() = BasicPanel(this)
    }
}